app/src/main/java/com/example/weatherapp/MainActivity.kt

```diff
@@ -1,11 +1,93 @@
 package com.example.weatherapp
 
+import android.Manifest
+import android.content.*
+import android.content.pm.PackageManager
+import android.os.Build
 import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
+import android.os.IBinder
+import androidx.core.content.ContextCompat
+import com.example.weatherapp.databinding.ActivityMainBinding
+import com.example.weatherapp.services.LocalizationService
 
 class MainActivity : AppCompatActivity() {
+
+    private lateinit var broadcastReceiver: BroadcastReceiver
+    private lateinit var mService: LocalizationService
+    private lateinit var binding: ActivityMainBinding
+    private var mBound: Boolean = false
+
+    private val connection = object : ServiceConnection {
+        override fun onServiceConnected(name: ComponentName, service: IBinder) {
+            val binder = service as LocalizationService.ServiceBinder
+            mService = binder.getService()
+            mBound = true
+        }
+
+        override fun onServiceDisconnected(componentName: ComponentName) {
+            mBound = false
+        }
+    }
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
+        binding = ActivityMainBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        if(!checkPermissions())
+            enableButton()
```

Comment:
What happens when a button is clicked and permission is not granted? Maybe we should inform the user that we need these permissions

Reply:
User is asked for permissions again if it is not granted. It is implemented in onRequestPermissionsResult. Maybe Toast message for the user would be good?

Reply:
both of those 2 options are fine:
-show toast
-request permission from user

Reply:
Additionaly added toast message